diff --git a/parser.c b/parser.c
index eb3d1d48..1bfcacb4 100644
--- a/parser.c
+++ b/parser.c
@@ -9593,7 +9593,7 @@ next_attr:
 	            if (j <= nbNs) continue;
 
 		    nsname = xmlGetNamespace(ctxt, attname);
-		    if (nsname != defaults->values[2]) {
+		    if (nsname != defaults->values[5 * i + 2]) {
 			if (nsPush(ctxt, attname,
 			           defaults->values[5 * i + 2]) > 0)
 			    nbNs++;
